packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/tags/GetTagsTests.java

```diff
@@ -226,4 +228,91 @@ public void checkThatDeletedTagIsNotRetrievedAnymore() throws Exception
                 .and().entriesListDoesNotContain("tag", removedTag.toLowerCase());
     }
+
+    /**
+     * Get tags with names filter using EQUALS and expect one item in result.
+     */
+    @Test(groups = { TestGroup.REST_API, TestGroup.TAGS, TestGroup.REGRESSION })
+    public void testGetTags_withSingleNameFilter()
+    {
+        STEP("Get tags with names filter using EQUALS and expect one item in result");
+        returnedCollection = restClient.authenticateUser(adminUserModel)
+            .withParams("where=(name='" + documentTag.getTag() + "')")
+            .withCoreAPI()
+            .getTags();
+
+        restClient.assertStatusCodeIs(HttpStatus.OK);
+        returnedCollection.assertThat()
+            .entriesListCountIs(1).and()
+            .entriesListContains("tag", documentTagValue.toLowerCase());
+    }
+
+    /**
+     * Get tags with names filter using IN and expect two items in result.
+     */
+    @Test(groups = { TestGroup.REST_API, TestGroup.TAGS, TestGroup.REGRESSION })
+    public void testGetTags_withTwoNameFilters()
+    {
+        STEP("Get tags with names filter using IN and expect two items in result");
+        returnedCollection = restClient.authenticateUser(adminUserModel)
+            .withParams("where=(name IN ('" + documentTag.getTag() + "', '" + folderTag.getTag() + "'))")
+            .withCoreAPI()
+            .getTags();
+
+        restClient.assertStatusCodeIs(HttpStatus.OK);
+        returnedCollection.assertThat()
+            .entriesListCountIs(2).and()
+            .entriesListContains("tag", documentTagValue.toLowerCase()).and()
+            .entriesListContains("tag", folderTagValue.toLowerCase());
+    }
+
+    /**
+     * Get tags with names filter using EQUALS (in fact contains) and expect two item in result.
+     */
+    @Test(groups = { TestGroup.REST_API, TestGroup.TAGS, TestGroup.REGRESSION })
+    public void testGetTags_whichNamesStartsWithOrphan()
+    {
+        STEP("Get tags with names filter using EQUALS (in fact contains) and expect two item in result");
+        returnedCollection = restClient.authenticateUser(adminUserModel)
+            .withParams("where=(name=orphan)")
+            .withCoreAPI()
+            .getTags();
+
+        restClient.assertStatusCodeIs(HttpStatus.OK);
+        returnedCollection.assertThat()
+            .entriesListIsNotEmpty().and()
+            .entriesListContains("tag", orphanTag.getTag().toLowerCase());
+    }
+
+    /**
+     * Try to get tags with names filter using EQUALS and wrong property name and expect 400 (Bad Request).
+     */
+    @Test(groups = { TestGroup.REST_API, TestGroup.TAGS, TestGroup.REGRESSION })
+    public void testGetTags_withWrongWherePropertyNameAndExpect400()
+    {
+        STEP("Try to get tags with names filter using EQUALS and wrong property name and expect 400");
+        returnedCollection = restClient.authenticateUser(adminUserModel)
+            .withParams("where=(tag=gat)")
```

Comment:
This is a good point. Do you think `where=(tag=...)` would be clearer than `where=(name=...)` for the client? Maybe we should propose this change in the API explorer?

Reply:
Yes, `tag` in where clause will be better as in result entry we are also returning field 'tag'.